magicctapipe/scripts/lst1_magic/muon_analysis_LST_or_MAGIC_data.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+Author: Gabriel Emery
+
+This script processes MAGIC calibrated data (*_Y_*.root) to perform the muon ring analysis.
+Current naming allows LST but first implementation will only handle MAGIC as script for LST are available in lstchain.
+
+Usage:
+$ python muon_analysis_LST_or_MAGIC_data.py
+--input-file ./data/calibrated/20201216_M1_05093711.001_Y_CrabNebula-W0.40+035.root
+--output-dir ./data/muons
+--config-file ./config.yaml
+"""
+import argparse
+import yaml
+import logging
+import numpy as np
+from pathlib import Path
+
+from astropy.table import Table
+from ctapipe_io_magic import MAGICEventSource
+from lstchain.image.muon import create_muon_table
+from magicctapipe.image import MAGICClean
+from magicctapipe.image.muons import perform_muon_analysis
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+pedestal_types = [
+    'fundamental',
+    'from_extractor',
+    'from_extractor_rndm',
+]
+
+
+def magic_muons_from_cal(input_file, output_dir, config, process_run):
+    """
+    Process all event a single telescope MAGIC calibrated data run or subrun to perform the muon ring analysis.
+
+    Parameters
+    ----------
+    input_file:
+    output_dir:
+    config:
+    process_run:
+
+    """
+
+    event_source = MAGICEventSource(input_url=input_file)
```

Comment:
I think it should work fine if you change to general EventSource, when the file is a MAGIC _Y_ file it will automatically select MAGICEventSource, and this will make it easier to run the same script also on LST